src/lib.rs

```diff
@@ -65,13 +75,26 @@ impl<'a> Toipe {
     /// Initializes the word selector.
     /// Also invokes [`Toipe::restart()`].
     pub fn new(config: ToipeConfig) -> Result<Self, ToipeError> {
+        let word_selector_error_prefix = "Error reading the given word list: ";
         let word_selector: Box<dyn WordSelector> =
             if let Some(wordlist_path) = config.wordlist_file.clone() {
-                Box::new(RawWordSelector::from_path(PathBuf::from(wordlist_path))?)
+                let word_selector_result = RawWordSelector::from_path(PathBuf::from(wordlist_path));
+                match word_selector_result {
+                    Ok(rws) => Box::new(rws),
+                    Err(e) => return Err(ToipeError::from(e).with_context(word_selector_error_prefix)),
+                }
             } else if let Some(word_list) = config.wordlist.contents() {
-                Box::new(RawWordSelector::from_string(word_list.to_string())?)
+                let word_selector_result = RawWordSelector::from_string(word_list.to_string());
+                match word_selector_result {
+                    Ok(rws) => Box::new(rws),
+                    Err(e) => return Err(ToipeError::from(e).with_context(word_selector_error_prefix)),
+                }
             } else if let BuiltInWordlist::OS = config.wordlist {
-                Box::new(RawWordSelector::from_path(PathBuf::from(OS_WORDLIST_PATH))?)
+                let word_selector_result = RawWordSelector::from_path(PathBuf::from(OS_WORDLIST_PATH));
+                match word_selector_result {
+                    Ok(rws) => Box::new(rws),
+                    Err(e) => return Err(ToipeError::from(e).with_context(word_selector_error_prefix)),
+                }
```

Comment:
There is quite a bit of repetition here, there must be a better way, but I'm not sure yet. I will get back to you on this.

Reply:
I tried to make this better in a30f43fa636198c51c8f78f14f2ecc299189513c. Take a look.